### ISSUE TITLE
How to get average loss across all ranks using custom loss function

### ISSUE THREAD

**Issue by mhy9989** [microsoft/DeepSpeed#4472]
Below is a simple training code
`for i, (inputs, labels) in enumerate(trainloader):
        inputs = inputs.to(device)
        labels = labels.to(device)
        pred = model(inputs)
        loss = criterion(pred, labels)
        optimizer.zero_grad()
        model.backward(loss)
        optimizer.step()`
In this code, although the gradient descent of the model can be performed automatically.
But the loss is different in each rank. So how do I get the average loss under all ranks?
Thank you.

**Comment by mhy9989** [comment]
> Below is a simple training code 
`for i, (inputs, labels) in enumerate(trainloader): 
inputs = inputs.to(device)  
labels = labels.to(device)  
pred = model(inputs)  
loss = criterion(pred, labels)  
optimizer.zero_grad()  
model.backward(loss)  
optimizer.step()  ` In this code, although the gradient descent of the model can be performed automatically. But the loss is different in each rank. So how do I get the average loss under all ranks? Thank you.
